### PR TITLE
Fixes KA-LSR

### PR DESCRIPTION
**Pull request by khizirsiddiqui**
Addresses a part of #56 .

@NeelayS I changed some part of implementation and it differs from yours, please point out if I missed something.

- Benchmarking Left

### CODE REVIEW

**Review by NeelayS — COMMENTED** [review]
Ah nice, a simple `torch.no_grad()` works. I did this whole workaround because direct assignment for the incorrect labels was breaking autograd, completely forgot about using `torch.no_grad()`

**Comment by NeelayS** [comment]
KA-PS can be simplified similarly then

**Review by Het-Shah — APPROVED** [review]
